### ISSUE TITLE
Issue with GPU code

### ISSUE THREAD

**Issue by AbandiGa** [chenxuhao/GraphMiner#1]
Hello,

I'm having an issue with the GPU code, when I run even with the graph provided I always get zero for the number of triangles or other patterns. The OpenMP code seems to work fine with me. I provided the output of both GPU and OpenMP.

GPU:

./tc_gpu_base /home/ghadeer/cit-Patents/graph
Triangle Counting (for undirected graphs only)
Using DAG (static orientation)
Orientation enabled, using DAG
Time on generating the DAG: 0.216688 sec
|V| 3774769 |E| 16518947
Found 4 devices
  Device[0]: NVIDIA TITAN V
  Compute capability: 7.0
  Warp size: 32
  Total # SM: 80
  Total # CUDA cores: 5120
  Total amount of shared memory per block: 49152 bytes
  Total # registers per block: 65536
  Total amount of constant memory: 65536 bytes
  Total global memory: 11.8 GB
  Memory Clock Rate: 0.81 GHz
  Memory Bus Width: 3072 bits
  Peak Memory Bandwidth: 652.80 GB/s

  Device[1]: NVIDIA GeForce RTX 2070 SUPER
  Device[2]: NVIDIA TITAN Xp
  Device[3]: Tesla K40c
GPU_total_mem = 12652838912 graph_mem = 162349736
Time on generating the edgelist: 0.070154 sec
Edge parallel: edgelist size = 16518947
max_blocks_per_SM = 0
CUDA triangle counting (65536 CTAs, 256 threads/CTA)
runtime [gpu_base] = 2e-06 sec
throughput = 8259.47 billion Traversed Edges Per Second (TEPS)
total_num_triangles = 0

OpenMP (which seems to work fine)

 ./tc_gpu_base 0 /home/ghadeer/cit-Patents/graph
Triangle Counting (for undirected graphs only)
Using DAG (static orientation)
tc_gpu_base: ../common/graph.cc:8: Graph::Graph(std::string, bool, bool): Assertion `f_meta' failed.
Aborted (core dumped)
[gaa54@ithaca bin]$ ./tc_gpu_base
Usage: ./tc_gpu_base <graph> [num_gpu(1)] [chunk_size(1024)]
Example: ./tc_gpu_base /graph_inputs/mico/graph
[gaa54@ithaca bin]$ ./tc_omp_base /home/ghadeer/cit-Patents/graph 0
Triangle Counting (for undirected graphs only)
Using DAG (static orientation)
Orientation enabled, using DAG
Time on generating the DAG: 0.215716 sec
|V| 3774769 |E| 16518947
Launching OpenMP TC solver (32 threads) ...
runtime [omp_base] = 0.241254 sec
total_num_triangles = 7515023

Thank you

**Comment by chenxuhao** [comment]
Have you set this flag for your GPU?

CUDA_ARCH := -gencode arch=compute_70,code=sm_70

**Comment by AbandiGa** [comment]
Yes, I set this flag for my GPU but I still have the same problem.

**Comment by chenxuhao** [comment]
> Yes, I set this flag for my GPU but I still have the same problem.

which GPU do you use? I don't see this problem with a V100.
